src/tree/EnvironmentTreeItem.ts

```diff
@@ -92,7 +89,7 @@ export class EnvironmentTreeItem extends AzureParentTreeItem implements IAzureRe
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean, _context: IActionContext): Promise<AzExtParentTreeItem[]> {
-        return [this.actionsTreeItem, this.appSettingsTreeItem, this.functionsTreeItem];
+        return [this.actionsTreeItem, this.functionsTreeItem];
```

Comment:
I think you're confusing "function"/"function app"/"trigger". If you were going to do the "overkill" tree structure, this is how I would word it:

> Function App
> -> Application Settings
> ---> test = value
> -> Functions
> ---> HttpTrigger1

I would prefer the existing UI or the slightly "overkill" UI over the proposed changes in this PR, which I think is even more confusing.

Another option (not sure if this is possible) would be to detect if the app has any functions deployed to it ahead of time. If not, hide both "functions" and "application settings" and instead show a generic node like "Add a function..."/"Deploy a function app..."/"Learn how to add functions..."

Reply:
It's definitely possible, it'll just take be slightly longer to load the build's children (from instant to an Azure API get call).

As far as the wording, I think the portal purposely avoids using the word Function App because you don't have full control over it like you would if you were actually using a function app proper.

![image](https://user-images.githubusercontent.com/5290572/98606034-47836200-229b-11eb-88bb-2132717f6f79.png)

I think since the API call is on the build itself (and not when you're loading the children of the subscription) I'm okay with the latency there.

Reply:
I played with the idea of putting a tree item for creating a function if the user didn't have one, but I didn't really like how it looked.

Also, we can't actually _create_ a function app for them.  Even if they do have their project opened in the local workspace, we'd still have to commit/push the repo for them.  I guess we could take them to this page though https://docs.microsoft.com/en-us/azure/static-web-apps/apis

![image](https://user-images.githubusercontent.com/5290572/98989991-bbfa1300-24de-11eb-8c9b-72b02a314bb4.png)

Reply:
> it'll just take be slightly longer to load the build's children (from instant to an Azure API get call).

This would be my biggest concern as well and it definitely needs to be considered. As you mentioned, though, it's probably okay assuming it only happens for one build at a time when you expand it.

>  but I didn't really like how it looked.

I think it looks weird because there's no icon - just add an icon.

> Also, we can't actually create a function app for them. Even if they do have their project opened in the local workspace, we'd still have to commit/push the repo for them. I guess we could take them to this page though https://docs.microsoft.com/en-us/azure/static-web-apps/apis

I'm fine with just a learn more link if that's easier

> Oh yeah, but they also don't nest Application Settings in Functions either, but they also have a terrible error message that's super hard to notice as well, not really sure it's a great pattern to follow

First of all - not sure why you think that error message is so terrible... Secondly, nesting app settings under functions is the only option I don't like. It's just not consistent with other UI (in the portal or the functions extension) and the labeling doesn't makes sense to me. If you don't want to do any of the other options, I'm totally fine keeping the UI as-is (with app settings and functions at the same level).

Reply:
Well, it's more that the message is hard to see (it's just a small ribbon at the top), but all the UI still functions so it's hard to notice that it's not actually doing anything.  It's not consistent with the other UI, but also Functions/App Service app settings affect the entire app whereas App Settings only applies to the Function app that lives in the SWA.  The portal actually calls it "Configuration" which I don't think is specific enough to help the user, but what if we called it Function Settings instead then?

Reply:
I'm not a big fan of inventing new terminology for platform features. The portal might use "Configuration", but [plenty of docs](https://docs.microsoft.com/azure/static-web-apps/application-settings) and our other extensions use "Application Settings"

Reply:
Okay, then I'll leave everything the same but just have a learn more about Azure Functions in Static Web Apps... node or something that directs them the doc you just linked.  Seem 👌?